tracing-appender/src/rolling.rs

```diff
@@ -0,0 +1,178 @@
+use crate::inner::InnerAppender;
+use chrono::{DateTime, Datelike, TimeZone, Timelike, Utc};
+use std::fmt::Debug;
+use std::fs::{File, OpenOptions};
+use std::io::{BufWriter, Write};
+use std::path::Path;
+use std::sync::{Mutex, MutexGuard};
+use std::{fs, io};
+use tracing_subscriber::fmt::MakeWriter;
+
+pub struct RollingFileAppender {
+    inner: Mutex<InnerAppender<BufWriterFactory>>,
```

Comment:
It's a shame that when this is used by the non-blocking writer, we still wrap the appender in a mutex, even though it's only used by the worker thread. That makes me wonder if making the `non_blocking` constructor take a `MakeWriter` was really the right approach...I think it should probably take a `T: Write` rather than a `T: MakeWriter`. Then, there could be a `Write` impl for `RollingFileAppender`, and if users want to use the rolling appender _without_ the `non_blocking` writer, it could be wrapped in a `Mutex` (and I'll add a `MakeWriter` impl for `Mutex<T> where T: Write`).

This also solves the issues around whether or not `make_writer` should get called multiple times. Sorry I gave you a bad suggestion earlier.

Reply:
This solution worked wonderfully! Using `T:Write` made everything much simpler